packages/conventional-changelog-core/lib/merge-config.js

```diff
@@ -53,10 +53,19 @@ function guessNextTag (previousTag, version) {
   return version
 }
 
+function omitUndefinedValueProps (obj) {
+  if (obj === null || typeof obj !== 'object') {
+    return {}
+  }
+  const omittedObj = Object.entries(obj).filter(([, value]) => value !== undefined)
+  return Object.fromEntries(omittedObj)
+}
+
```

Comment:
let's optimize it to one loop

```suggestion
function omitUndefinedValueProps (obj) {
  if (!obj) {
    return {}
  }

  const omittedObj = {}
  
  for (const key in obj) {
    if (obj[key] !== undefined) {
      omittedObj[key] = obj[key]
    }
  }
  
  return omittedObj
}

```

Reply:
Optimized in [8f118d5](https://github.com/conventional-changelog/conventional-changelog/pull/1018/commits/8f118d5ee0da625e741012a08f5d8e4f235857f7)